---
title: 'Some tips when working with PLINK2'
date: 2020-07-02
permalink: /posts/2020/09/PLINK2
tags:
  - computing
---

PLINK is a well-established software for genetic analysis. In many project, we use `plink2` for genome-wide association study (GWAS) and other computations related to the raw genotype matrix. Here, I list several tips on the use of this software.

## `pgenlibr` to read plink2 pgen files from `R`

To read genotype data stored in `plink2` pgen/pvar/psam files, one can use an R package called `pgenlibr`. Here I show some example usage of `pgenlibr`.

### How to install `pgenlibr` in your R environment?

```{bash}
$ git clone https://github.com/chrchang/plink-ng.git
$ R
> install.packages('plink-ng/2.0/pgenlibr', repos = NULL, type='source')
```

### Some basic usage of `pgenlibr`

```{R}
require(pgenlibr)

pfile <- '@@@/sample_genotype_data'

pvar <- NewPvar(paste0(pfile, '.pvar.zst'))
pgen <- NewPgen(paste0(pfile, '.pgen'), pvar=pvar)

GetVariantId(pvar, 1)
GetVariantId(pvar, 10)
GetVariantCt(pvar)
GetRawSampleCt(pgen)

# read a single variant
buf <- pgenlibr::Buf(pgen)
pgenlibr::Read(pgen, buf, 1)

# read a list of variants
## var.idxs (list of variants you would like to read)
geno_mat <- ReadList(pgen, var.idxs, meanimpute=F)
```

It seems like the package is still an early development version without a full documentation.
You can see the list of supported functions in its [source code on GitHub](https://github.com/chrchang/plink-ng/blob/master/2.0/pgenlibr/R/RcppExports.R).

#### How to convert your genetic data into plink format

The pgenlibr library works with [plink2's pgen/pvar/psam format](https://github.com/chrchang/plink-ng/blob/master/pgen_spec/pgen_spec.pdf).
You can use `plink2` binary software to conver yout genetic dataset into this format. For example, the following [`--make-pgen` (link to reference)](https://www.cog-genomics.org/plink/2.0/data#make_pgen) command will convert your `input.vcf` file into `binary_fileset.{pgen,pvar,psam}`. Many [other input formats](https://www.cog-genomics.org/plink/2.0/input) are also supported in plink2.

```{bash}
plink2 --vcf input.vcf --make-pgen --out binary_fileset
```

## Running recessive model on chrX

As illustrated in PLINK2 website, `--glm recessive` will be the modifier to run GWAS scan with recessive model. However, this does not work well for X chromosome (chrX). To mitigate this limitation, we can record the chromosome X in the pvar file and pretend as if another autosome first and run thee association analysis.

1. use `--output-chr 26` to change the `.pvar` to encode chrX as "23"
2. run the original `--glm` command with e.g. `--chr-set 40` (which specifies 40 instead of 22 autosomes).

## GWAS of multiple quantitative traits

Starting `plink2/20190401`, the program supports efficient linear regression of multiple quantitative phenotypes.

> --glm without --adjust now detects groups of quantitative phenotypes with the same "missingness pattern", and processes them together

One should use this feature for efficient computation.

## Reference

- [PLINK 2.0 website](https://www.cog-genomics.org/plink/2.0/)
- [PLINK 2.0 GitHub / pgenlibr](https://github.com/chrchang/plink-ng/tree/master/2.0/pgenlibr)
- [The pgen/pvar/psam format specification](https://github.com/chrchang/plink-ng/blob/master/pgen_spec/pgen_spec.pdf)
