---
title: "WhichTF is dominant in your open chromatin data?"
collection: publications
permalink: /publication/2022-08-30-whichtf
excerpt: "We develop an ontology-guided approach to ranking tissue-/cell-type-specific transcription factors (TFs) from chromatin accessibility data."
excerpt_image: '/files/2022/WhichTF-Fig1.png'
date: 2022-08-30
venue: 'PLOS Comput Biol'
paperurl: 'https://doi.org/10.1371/journal.pcbi.1010378'
citation: 'Y. Tanigawa*, E. S. Dyer*, G. Bejerano, WhichTF is dominant in your open chromatin data?  PLOS Comput Biol 18(8): e1010378. (2022).'
---

We develop an ontology-guided approach to ranking tissue-/cell-type-specific transcription factors (TFs) from chromatin accessibility data.

![WhichTF Fig. 1](/files/2022/WhichTF-Fig1.png)

## Highlights

Transcription factors (TFs) regulate context-specific gene expression. Genome-wide chromatin accessibility profile coupled with a computational analysis has been one of the most promising approaches to rank TFs enriched in a given sample. Most existing methods rely on statistical enrichment, either on sequence motifs or occurrences of the TF binding sites. We empirically find that those abundance-based approaches tend to be less cell-type-specific. For example, CTCF is frequently ranked among the top hits.

To address this, we propose an alternative, WhichTF, that aggregates the results of ontology-based stratified enrichment tests to rank TFs. Our model considers ontology-based annotations of genomic regions on top of high-confidence prediction of conserved TF binding sites. Specifically, we decompose the user-provided accessible profile into a series of genomic tracks annotated for ontology terms. We consider the most enriched terms in the accessible peaks as a proxy to capture cellular contexts. Aggregating across terms, we score and rank TFs.

![WhichTF Fig. 1](/files/2022/WhichTF-Fig1.png)

When applied to well-characterized datasets, we find the cell-type-specific roles of the top-ranked TFs are supported by the literature.

![WhichTF Table 1](/files/2022/WhichTF-Table1.png)

Applying across ~90 human samples followed by the t-SNE projection of the WhichTF scores TFs indicates WhichTF captures biological similarities and dissimilarities of TF-mediated transcriptional programs.

![WhichTF Fig. 2](/files/2022/WhichTF-Fig2.png)

To investigate closely related cells, we present differential TF ranking with WhichTF, where we apply WhichTF on differentially accessible regions. The top-ranked TFs are well-supported by the literature.

![WhichTF Table 2](/files/2022/WhichTF-Table2.png)

Beyond well-characterized cell types, we present applications in mesoderm development,

![WhichTF Fig. 3](/files/2022/WhichTF-Fig3.png)

... and patient-derived samples for systemic lupus erythematosus.

![WhichTF Table 3](/files/2022/WhichTF-Table3.png)

Overall, we show that ontology-guided stratified enrichment is a powerful approach to ranking cell-type-specific TFs. We highlight examples in human and mouse cells, developmental trajectories, and disease samples.

This work was started and was jointly led by Ethan S. Dyer and myself while I was in Prof. Gill Bejeranoâ€™s lab at Stanford, with tremendous support and help from many lab members. I want to thank Gill, Ethan, and my colleagues. Thank you, all!

## Resources

WhichTF software and reference data are available on BitBucket and figshare.

- [Program (Bitbucket)](https://bitbucket.org/bejerano/whichtf)
- [Reference Data (figshare)](https://doi.org/10.6084/m9.figshare.12774539)
- [Paper](https://doi.org/10.1371/journal.pcbi.1010378)

Also, as a part of this project, we performed a v4 update of [the Genomic Regions Enrichment of Annotations Tool (GREAT)](http://great.stanford.edu/).
