---
title: "Large-scale genomic inference of multiple phenotypes"
excerpt: ''
excerpt_image: ''
collection: talks
type: "Talk"
permalink: /talks/2021-05-03-PhD-defense
venue: "Stanford University, Department of Biomedical Data Science"
date: 2021-05-03
location: "Zoom"
---
<!-- paperurl: ''
citation: ''
tags: -->

![Celebration with the committee](/files/2021/20210503_committee.jpg)

## Yosuke passed the thesis defense!

On 2021/5/3, I presented my thesis titled "Large-scale genomic inference of multiple phenotypes" and passed my thesis defense. Thank you very much for everyone who supported me along the way.

## Abstract

Many human traits are multifactorial. Some of them have shared genetic basis, yet a systematic analysis of such components across the human phenome has been challenging, limiting our understanding of the shared genetic factors across traits and their influences on diseases. I will present four examples of multi-trait analysis, which I was fortunate to lead in my graduate studies.

We developed DeGAs (decomposition of genetic associations), systematically characterized the latent components of genetic associations across more than 2,000 phenotypes in UK Biobank, and demonstrated its applications to study adipocyte biology[1].

Protein-altering variants that are protective against human disease provide in vivo validation of therapeutic targets. Investigating multiple phenotypic endpoints in the datasets consisting of more than 514,000 European individuals in the UK and Finland, we identified rare protein-altering variants in ANGPTL7 that reduce glaucoma risk[2].

In a comprehensive genetic analysis of 35 blood and urine biomarkers in UK Biobank[3], we (i) characterized the genetic variations of serum and urine laboratory tests; (ii) built predictive polygenic risk score (PRS) models for each biomarker using genotypes; and (iii) demonstrated their influences on diseases.

In the last example, we consider the analysis of multiple molecular phenotypes, specifically focusing on transcription factors (TFs). To identify TFs with cell-type-specific functional importance from experimentally characterized open chromatin regions, I co-developed WhichTF, where we integrate functional annotation in ontology, sequence conservation, and gene regulatory domain models[4].

I will conclude my presentation by summarizing my contributions and the exciting research opportunities in the field.

### References

1. Y. Tanigawa\*, J. Li\*, et al., Nat Commun. 10, 4064 (2019). [PMID: 31492854](https://pubmed.ncbi.nlm.nih.gov/31492854/)
2. Y. Tanigawa, et al., PLOS Genetics. 16, e1008682 (2020). [PMID: 32369491](https://pubmed.ncbi.nlm.nih.gov/32369491/)
3. N. Sinnott-Armstrong\*, Y. Tanigawa\*, et al., Nat Gen. (2021). [PMID: 33462484](https://pubmed.ncbi.nlm.nih.gov/33462484/)
4. Y. Tanigawa\*, E. S. Dyer\*, G. Bejerano, bioRxiv, 730200 (2019). [doi:10.1101/730200](https://doi.org/10.1101/730200)
